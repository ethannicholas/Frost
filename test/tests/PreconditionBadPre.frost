@extendable
class PreconditionBadPre1 {
    @extendable
    @pre(x > 0)
    method test(x:Int) {
    }
}

class PreconditionBadPre2 : PreconditionBadPre1 {
    @override
    @pre(x < -10)
    method test(x:Int) {
    }
}