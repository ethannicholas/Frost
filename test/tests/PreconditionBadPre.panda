class PreconditionBadPre1 {
    @class
    @pre(x > 0)
    method test(x:Int) {
    }
}

class PreconditionBadPre2 : PreconditionBadPre1 {
    @class
    @override
    @pre(x < -10)
    method test(x:Int) {
    }
}