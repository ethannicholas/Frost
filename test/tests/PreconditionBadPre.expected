PreconditionBadPre.frost:11:12: error: cannot use '@pre' in a method which overrides another method; did you mean '@preOr'?
1 error
